server/game/cards/09-HoT/WillasTyrell.js

```diff
@@ -0,0 +1,64 @@
+const DrawCard = require('../../drawcard.js');
+
+class WillasTyrell extends DrawCard {
+    setupCardAbilities() {
+        this.action({
+            title: 'Kneel Willas and stand/give keywords',
+            condition: () => !this.kneeled,
+            handler: () => {
+                let partials = [];
+
+                this.controller.kneelCard(this);
+
+                let margaery = this.controller.findCardByName(this.controller.cardsInPlay, 'Margaery Tyrell');
+                if(margaery && margaery.kneeled) {
+                    this.controller.standCard(margaery);
+                    partials.push('stands {2}');
+                }
+
+                let flowers = this.controller.findCardByName(this.controller.cardsInPlay, 'The Knight of Flowers');
+                if(flowers) {
+                    this.untilEndOfPhase(ability => ({
+                        match: flowers,
+                        effect: ability.effects.addKeyword('insight')
+                    }));
+                    partials.push('gives insight to {3}');
+                }
+
+                let garlan = this.controller.findCardByName(this.controller.cardsInPlay, 'Ser Garlan Tyrell');
+                if(garlan) {
+                    this.untilEndOfPhase(ability => ({
+                        match: garlan,
+                        effect: ability.effects.addKeyword('renown')
+                    }));
+                    partials.push('gives renown to {4}');
+                }
+
+                let formattedPartials = this.formatPartials(partials);
+                this.game.addMessage('{0} uses {1} to kneel {1}' + formattedPartials, this.controller, this, margaery, flowers, garlan);
+            }
+        });
+    }
+
+    formatPartials(partials) {
```

Comment:
The existence of this function bothers me a great deal, as it indicates a major deficiency in the message system. But I'll let it through for now and try to fix it and similar cards (Dornishman's Wife) separately.